packages/core/test/signal.test.tsx

```diff
@@ -286,6 +286,48 @@ describe("effect()", () => {
 		expect(spy).not.to.be.called;
 	});
 
+	it("should dispose of subscriptions", () => {
+		const a = signal("a");
+		const b = signal("b");
+		const spy = sinon.spy(() => {
+			a.value + " " + b.value;
+		});
+		effect(function () {
+			spy();
+			if (a.value === "aa") {
+				this.dispose();
+			}
+		});
+
+		expect(spy).to.be.called;
+
+		a.value = "aa";
+		expect(spy).to.be.calledTwice;
+
+		a.value = "aaa";
+		expect(spy).to.be.calledTwice;
+	});
+
+	it("should dispose of subscriptions immediately", () => {
+		const a = signal("a");
+		const b = signal("b");
+		const spy = sinon.spy(() => {
+			a.value + " " + b.value;
+		});
+		effect(function () {
+			spy();
+			this.dispose();
+		});
+
+		expect(spy).to.be.calledOnce;
+
+		a.value = "aa";
+		expect(spy).to.be.calledOnce;
+
+		a.value = "aaa";
+		expect(spy).to.be.calledOnce;
+	});
+
```

Comment:
Nit: Let's add a test where `dispose()` is called from inside the callback and after that the returned dispose function is called. Just to ensure that we're not erroring when it's called twice